### PR TITLE
fix: wc to dApps shenanigans

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

A story in many parts involving a partial loss of sanity, weird specs and a not-so-lovely debugging experience.

1. handles `wallet_addEthereumChain` and `wallet_switchEthereumChain` as valid events. No, we don't handle "adding" or switching chains for wc, however certain dApps will spam those session requests to wc and failure to handle those messages means dApps will assume wrong network connected and will never allow continuing, discretely failing
2. Now that we handle those messages, responds with happy JSON-RPC response to wc so that clients assume those requests went fine and we "added" or "switched" chains
3. handles *all* gas params missing, which *is* a valid message for all intents and purposes
4. handles Tx value in hex and stringifies it
5. handles native asset fees denominated in base unit, which we didn't do previously

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/6488

Low but not none:

- Responding with happy responses to `wallet_addEthereumChain` and `wallet_switchEthereumChain` means some dApps may wrongly assume chain support (shouldn't be the case given capabilities are clearly defined as part of namespaces, but...)
- The assumption here is `value` was wrong all along and should've been converted from native assets base unit. If `value` was ever sent as crypto precision, then we would effectively use a wrong exponent here.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- ether.fi Tx signing is happy

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

https://github.com/shapeshift/web/assets/17035424/b7b453b5-4eee-46b4-bfcb-5754f3d82315

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6572?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6572** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6572?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6553** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6553?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/plugins/walletConnectToDapps/components/modals/EIP155TransactionConfirmation.tsx [hunk 1]
```diff
@@ -94,6 +96,17 @@ export const EIP155TransactionConfirmation: FC<
     [fees, feeAsset, feeAssetPrice],
   )
 
+  const value = useMemo(() => {
+    if (!feeAsset) return '0'
+
+    const valueCryptoBaseUnit =
+      transaction?.value && isHex(transaction.value)
```
Comment (gomesalexandre) [inline src/plugins/walletConnectToDapps/components/modals/EIP155TransactionConfirmation.tsx:103]: Handles value as hex string, which is valid

File: src/plugins/walletConnectToDapps/components/modals/EIP155TransactionConfirmation.tsx [hunk 2]
```diff
@@ -94,6 +96,17 @@ export const EIP155TransactionConfirmation: FC<
...
+        ? fromHex(transaction.value, 'bigint').toString()
+        : transaction?.value
+    const valueCryptoPrecision = fromBaseUnit(valueCryptoBaseUnit ?? '0', feeAsset.precision)
```
Comment (gomesalexandre) [inline src/plugins/walletConnectToDapps/components/modals/EIP155TransactionConfirmation.tsx:106]: `value` should always be denominated in wei and that was probably wrong the whole time

File: src/plugins/walletConnectToDapps/eventsManager/useWalletConnectEventsHandler.ts [hunk 3]
```diff
@@ -75,6 +76,15 @@ export const useWalletConnectEventsHandler = (
             },
           })
 
+        case EIP155_SigningMethod.WALLET_ADD_ETHEREUM_CHAIN:
+        case EIP155_SigningMethod.WALLET_SWITCH_ETHEREUM_CHAIN:
+          // This doesn't fit our usual pattern and doesn't require a modal
+          // We implicitly return a happy JSON-RPC response
```
Comment (gomesalexandre) [inline src/plugins/walletConnectToDapps/eventsManager/useWalletConnectEventsHandler.ts:82]: Does what it says on the comment - the guts of this modal being fired

File: src/plugins/walletConnectToDapps/typeGuards.ts [hunk 4]
```diff
@@ -55,5 +55,11 @@ export const isTransactionParams = (
   Boolean(
     (transaction.gasLimit !== undefined && transaction.gasPrice !== undefined) ||
       transaction.gas !== undefined ||
-      (transaction.maxFeePerGas !== undefined && transaction.maxPriorityFeePerGas !== undefined),
+      (transaction.maxFeePerGas !== undefined && transaction.maxPriorityFeePerGas !== undefined) ||
+      // All gas params undefined is also a valid payload
```
Comment (gomesalexandre) [inline src/plugins/walletConnectToDapps/typeGuards.ts:59]: Does what it says on the comment - the modal firing is a nice fix, but not so nice if it's in an infinite loading state - all gas params missing *is* valid

**Review by 0xApotheosis — APPROVED** [review]
✅ I was able to successfully sign a TX on ether.fi.

gm for the wallet connect package bump, too!

**Pull request merged**